comp/otelcol/otlp/integrationtest/integration_test.go

```diff
@@ -119,8 +119,8 @@ func runTestOTelAgent(ctx context.Context, params *subcommands.GlobalParams) err
 		}),
 		logsagentpipelineimpl.Module(),
 		// We create strategy.ZlibStrategy directly to avoid build tags
-		fx.Provide(strategy.NewZlibStrategy),
-		fx.Provide(func(s *strategy.ZlibStrategy) compression.Component {
+		fx.Provide(compressionimpl.NewComponent),
+		fx.Provide(func(s *compressionimpl.ZlibStrategy) compression.Component {
```

Comment:
Is this line needed? `ZlibStrategy` could probably not be exported anymore.

Reply:
This would change the behaviour so the collector will choose the compression strategy based on the configuration setting rather than always using `zlib` as before. Is that intended?